service/storage_proxy.cc

```diff
@@ -411,7 +412,7 @@ class storage_proxy::remote {
             netw::messaging_service::msg_addr src_addr, rpc::opt_time_point t,
             auto schema_version, auto in, inet_address_vector_replica_set forward, gms::inet_address reply_to,
             unsigned shard, storage_proxy::response_id_type response_id, std::optional<tracing::trace_info> trace_info,
-            auto&& apply_fn1, auto&& forward_fn1) {
+            fencing_token fence, auto&& apply_fn1, auto&& forward_fn1) {
```

Comment:
I hate this function

Reply:
Why? Not enough parameters?

Reply:
Too many autos

Reply:
It is possible to specify real types here, but the function definition will be even longer.

Reply:
Even worse, someone might be able to understand what's going on